internal/repo/update.go

```diff
@@ -0,0 +1,141 @@
+package repo
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/zu1k/nali/internal/constant"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/google/go-github/v55/github"
+)
+
+var (
+	ctx      = context.Background()
+	tAsset   *github.ReleaseAsset
+	shaAsset *github.ReleaseAsset
+)
+
+func UpdateRepo() error {
+	rel, err := getLatestRelease()
+	if err != nil {
+		return fmt.Errorf("failed to get latest release: %v", err)
+	}
+
+	if !canUpdate(rel) {
+		return fmt.Errorf("current version %v is greater or equal to the latest version %v, no update", constant.Version, rel.GetTagName())
+	}
```

Comment:
just `return nil` if current version greater or equal to the latest version, this is a common situation and does not need to be reported to the user

Reply:
And we should tell the user that nali is already the latest version